src/components/Modal/index.tsx

```diff
@@ -0,0 +1,34 @@
+import React, { FC, useEffect, useRef, useState } from 'react';
+import { useRefCloseOut } from '../../hooks/useRefCloseOut';
+import { Button } from '../UI/Button';
+
+import styles from './Modal.module.scss';
+
+export const Modal: FC<{ isActive: boolean; children; action?: string }> = ({ isActive, children, action }) => {
+  const [active, setActive] = useState(null);
+  const refModal = useRef() as React.MutableRefObject<HTMLDivElement>;
```

Comment:
ref правильнее так типизировать

```suggestion
  const refModal = useRef<HTMLDivElement>(null);
```